public/app/features/alerting/unified/hooks/useStateHistoryModal.tsx

```diff
@@ -1,32 +1,83 @@
-import React, { useMemo, useState } from 'react';
+import { css } from '@emotion/css';
+import React, { lazy, Suspense, useCallback, useMemo, useState } from 'react';
 
-import { Modal } from '@grafana/ui';
+import { GrafanaTheme2 } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { Modal, useStyles2 } from '@grafana/ui';
+import { RulerGrafanaRuleDTO } from 'app/types/unified-alerting-dto';
 
-import { StateHistory } from '../components/rules/StateHistory';
+const AnnotationsStateHistory = lazy(() => import('../components/rules/state-history/StateHistory'));
+const LokiStateHistory = lazy(() => import('../components/rules/state-history/LokiStateHistory'));
 
-function useStateHistoryModal(alertId: string) {
```

Comment:
`openModal` takes a `RulerGrafanaRuleDTO` as a parameter because for the old state view we need `rule.grafana_alert.id` whereas the new one needs **uid** (`rule.grafana_alert.uid`)
Hopefully we'll migrate the old panel to `uid` in the future